core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -341,6 +323,45 @@ public KeyStore getPrimaryKeyStore() throws Exception {
         }
     }
 
+    /**
+     * Load the requested tenant keystore from registry.
+     *
+     * @param keyStoreName  Name of the Tenant KeyStore.
+     * @return Key store object.
+     * @throws Exception    Exception if failed to retrive the given keystore from registry.
+     */
+    private KeyStore getTenantKeyStore(String keyStoreName) throws Exception {
+
+        if (isCachedKeyStoreValid(keyStoreName)) {
+            return tenantKeyStores.get(keyStoreName).getKeyStore();
+        }
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
+        if (registry.resourceExists(path)) {
+            org.wso2.carbon.registry.api.Resource resource = registry.get(path);
```

Comment:
It is possible to import `org.wso2.carbon.registry.api.Resource` at the top of the class?

Reply:
Unfortunately, no. `org.wso2.carbon.registry.core.Resource` has already been imported. So importing `org.wso2.carbon.registry.api.Resource` will raise complications.